decidim/decidim
# Additional GraphQL fixes

#### :tophat: What? Why?
This PR adds some additional GraphQL imrovements. 
1. Decouples Decidim::ParticipatoryProcessType to the engine instead of Core
2. Fixes a bug when single private spaces could not be found by admin
3. Fixes a bug when unpublished components could not be found via api by admin 

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #13511

#### Testing
1. Authentitate to admin 
2. Unpublish a component from a participatory space
3. As admin interogate API to fetch the components 
4. See there is an error 
5. Make the space private, see the admin cannot interogate via API
6. Apply patch & restart server 
7. See space is now visible 
8. See the component is visible 
9. In incognito, as not logged in user, repeat 3, 4 
10. See the space is not visible ( returns null )
11. Make the space public, and see in api it cannot be viewed by anonymous user 

```
{
  participatoryProcess(id: 2){
    id
    components {
      id
      name {
        translation(locale: "en")
      }
    }
  }
}
```

### :camera: Screenshots
*Please add screenshots of the changes you are proposing*
![Description](URL)

:hearts: Thank you!
